io/io/v7/src/TFile.cxx

```diff
@@ -97,7 +97,7 @@ class TV6Storage: public ROOT::Experimental::Internal::TFileStorageInterface {
   }
 
   void WriteMemoryWithType(std::string_view name, const void* address, TClass* cl) final {
-    fOldFile->WriteObjectAny(address, cl, name.to_string().c_str());
+     fOldFile->WriteObjectAny(address, cl, name.data());
```

Comment:
Isn't data() not guaranteed to be null terminated?  (which WriteObjectAny requires)?